lib/api_resources/root.js

```diff
@@ -27,29 +28,119 @@ RootResource.prototype.proxy = function(env, next) {
 RootResource.prototype.list = function(env, next) {
   var self = this;
   var params = env.route.query;
-  if(params.ql) {
+  if (params.ql) {
     var ql = params.ql;
     var server = params.server || this.server.id;
 
-    if(this.shouldProxy(server)) {
+    if (server === '*') {
+      this.server.peerRegistry.find(Query.of('peers'), function(err, peers) {
+        if (err) {
+          env.response.statusCode = 500;
+          return next(env);
+        }
+
+        var href = '/servers/{{peerName}}';
+        var qs ='?' + querystring.stringify({ql: params.ql});
+        env.request.templateUrl = href + qs;
+
+        var httpServer = self.server.httpServer;
+
+        httpServer.proxyToPeers(peers, env, function(err, results) {
+          var messageId;
+
+          var entities = results.map(function(obj) {
+            var response = obj.res;
+            var body = obj.body;
+
+            if (obj.err) {
+              return [];
+            }
+
+            var id = response.headers['zetta-message-id'];
+
+            if (!messageId) {
+              messageId = id;
+            }
+
+            var res = httpServer.clients[id];
+
+            if (!res) {
+              return [];
+            }
+
+            Object.keys(response.headers).forEach(function(header) {
+              if (header !== 'zetta-message-id') {
+                res.setHeader(header, response.headers[header]);
+              }
+            });
+
+            res.statusCode = response.statusCode;
+
+            if (body) {
+              body = JSON.parse(body.toString());
+              return body && body.entities ? body.entities : [];
+            } else {
+              return [];
+            }
+          }).reduce(function(prev, curr) {
+            return prev.concat(curr);
+          }, []);
+
+          self.server.runtime.registry.find(params.ql, function(err, results) {
+            var localEntities = (!!err || !results) ? [] : results.map(function(device){
+              var deviceOnRuntime = self.server.runtime._jsDevices[device.id];
+              if (deviceOnRuntime) {
+                var model = deviceOnRuntime;
+                var loader = {path: '/servers/' + encodeURI(self.server.id)}
+                var server = self.server;
+                var entity = {
+                  class: ['device'],
+                  rel: [rel.device],
+                  properties: model.properties(),
+                  links: [{ rel: ['self'], href: env.helpers.url.path(loader.path + '/devices/' + model.id) },
+                          { title: server._name, rel: ['up', rel.server], href: env.helpers.url.path(loader.path) }]
+                };
+                return entity;
+              }
+            }).filter(function(entity) { return entity !== null });
+
+            var res = httpServer.clients[messageId];
+            var obj = {
+              class: ['root', 'search-results'],
+              properties: {
+                ql: params.ql
```

Comment:
That's what i was kinda thinking, I just noticed the properties had `ql` and it made me question if server should be there as well. I could see it helping with api consumers creating a UI refilling inputs and what not.